django/db/models/expressions.py

```diff
@@ -925,6 +925,32 @@ def relabeled_clone(self, relabels):
         return self
 
 
+class Excluded(Expression):
+    def __init__(self, field_name):
+        super().__init__()
+        self.field_name = field_name
+
+    def resolve_expression(self, *args, **kwargs):
+        resolved = F(self.field_name).resolve_expression(*args, **kwargs)
+        self.output_field = resolved.output_field
+        return self
+
+    def as_sql(self, compiler, connection):
+        quoted_field_name = compiler.quote_name_unless_alias(self.field_name)
+        if connection.vendor in ["postgresql", "sqlite"]:
```

Comment:
I @felixxm  understand your concern about hard-coding vendor names. My primary goal is not to check if a backend supports Excluded, but to generate the correct SQL code depending on the backend for use cases like condition=Q(rank__gt=Excluded("number") + 1) & ~Q(name__startswith="A") | Q(number__gt=3).

I am very open to suggestions for improving this implementation. Perhaps we could consider a more abstract method in BaseDatabaseOperations, which can be adapted by each backend to handle Excluded appropriately, while maintaining a uniform and 'Django-esque' interpretation of Excluded across different backends. Your input on the best way forward would be highly valuable

Reply:
I @felixxm  understand your concern about hard-coding vendor names. My primary goal is not to check if a backend supports Excluded, but to generate the correct SQL code depending on the backend for use cases like condition=Q(rank__gt=Excluded("number") + 1) & ~Q(name__startswith="A") | Q(number__gt=3).

I am very open to suggestions for improving this implementation. Perhaps we could consider a more abstract method in BaseDatabaseOperations, which can be adapted by each backend to handle Excluded appropriately, while maintaining a uniform and 'Django-esque' interpretation of Excluded across different backends. Your input on the best way forward would be highly valuable

Reply:
> Please add a new feature flag.

Have you seen my comment? :point_up:

Reply:
> > Please add a new feature flag.
> 
> Have you seen my comment? ☝️

I really don't know if we understand each other. I'll suggest something in the next revision and you'll let me know.

Reply:
`Excluded` makes sense only for backend that support `supports_update_conflicts_with_condition`, so there is not need to check vendor names you should check `supports_update_conflicts_with_condition` flag in the `Excluded.as_sql()`, e.g.
```python
def as_sql(self, compiler, connection):
    if not connection.features.supports_update_conflicts_with_condition:
        raise NotSupportedError(...)
````

Reply:
> `Excluded` makes sense only for backend that support `supports_update_conflicts_with_condition`, so there is not need to check vendor names you should check `supports_update_conflicts_with_condition` flag in the `Excluded.as_sql()`, e.g.
> 
> ```python
> def as_sql(self, compiler, connection):
>     if not connection.features.supports_update_conflicts_with_condition:
>         raise NotSupportedError(...)
> ```

yes indeed. you're right. i hadn't thought of that. 
However, our discussions have made me rethink my approach to the question. But I'll make a note of it in case I need to keep that approach.